### PR TITLE
SWC-6529: publish @sage-bionetworks/synapse-types package

### PR DESCRIPTION
**Pull request by hallieswan**
See passing npmpublish dryrun for `@sage-bionetworks/synapse-types` [here](https://github.com/Sage-Bionetworks/synapse-web-monorepo/actions/runs/7415916429/job/20179937481)

Note: failing e2e tests are fixed in #634

### CODE REVIEW

File: .github/workflows/npmpublish.yml [hunk 1]
```diff
@@ -10,16 +10,26 @@ jobs:
...
+          ]
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3.1.0
       - uses: ./.github/actions/pnpm-setup-action
+      - name: Get package path
+        id: package-path
+        run: |
+          packagePath=$(pnpm nx show project ${{ matrix.publishablePackage }} --json | jq -r '.sourceRoot')
+          echo "packagePath=${packagePath}" >> "$GITHUB_OUTPUT"
```
Comment (hallieswan) [inline .github/workflows/npmpublish.yml:27]: Handle case where package name (`@sage-bionetworks/synapse-types`) may differ from package path (`packages/synapse-types`)

File: packages/synapse-types/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2023 Sage Bionetworks
```
Comment (hallieswan) [inline packages/synapse-types/LICENSE:189]: Happy to update if there's more appropriate license or copyright date to use here.

Comment (nickgros) [reply packages/synapse-types/LICENSE:189]: Apache 2 seems right, since we use that for other libs

**Review by nickgros — APPROVED** [review]
Looks good, let's see if it works 🙂

**Pull request merged**